lib/datasource/cdnjs/index.ts

```diff
@@ -27,38 +28,42 @@ async function downloadLibrary(library: string): CachePromise<CdnjsResponse> {
   return { data: (await http.getJson<CdnjsResponse>(url)).body };
 }
 
-export async function getReleases({
-  lookupName,
-}: GetReleasesConfig): Promise<ReleaseResult | null> {
-  // Each library contains multiple assets, so we cache at the library level instead of per-asset
-  const library = lookupName.split('/')[0];
-  try {
-    const { assets, homepage, repository } = await cacheAble({
-      id,
-      lookup: library,
-      cb: downloadLibrary,
-    });
-    if (!assets) {
-      return null;
-    }
-    const assetName = lookupName.replace(`${library}/`, '');
-    const releases = assets
-      .filter(({ files }) => files.includes(assetName))
-      .map(({ version, sri }) => ({ version, newDigest: sri[assetName] }));
+export class CdnJs extends Datasource {
```

Comment:
I like the class based idea, we should export the class as `default` or with a fixed alias, so we can generate the imports as we now do.

See `versioning` for sample. I think we need to undo the dynamic datasource loading to be like the other modules.

Reply:
I can export as default no problem.

What was the problem the dynamic generation/loading was trying to solve? I understand the alternative is to maintain the list in `lib/datasource/index.ts` manually, but this list doesn't really change that often.

Reply:
it was to do not have conflicts when users adding new versionigs / datasources / managers ...

Reply:
Oops. I didn't get a notification for your comment.

Is that such a common issue? I feel like it's a rare problem?

Reply:
No, shouldn't be a big problem.

@rarkins can / should we go back to static imports and remove the code generation?

Reply:
One remaining problem with maintaining a list was that it wouldn't support dynamic addition of datasources in future.

Reply:
As long as we have the function for returning the full datasource map, we can simply add new datasources.

You can see it in action [here](https://github.com/renovatebot/internal-tools/blob/15413767c5beffaebfe172fe72d1c2063b8ec157/src/utils/datasource/index.ts#L1-L9).
It's used by our internal tools. 🙃

Reply:
If we maintain a static list, I prefer to move it to a separate file, as it will be very long.

Reply:
I meant for end users to conveniently add custom datasources. If such custom ones need to be pre-configured in an index file, it makes it harder.

Reply:
If a user needs to add custom datasources i would suggest him to use the approach used by our internal-tools.